### PR TITLE
Complete food order

### PR DESCRIPTION
**Pull request by TNagaraja**
Issue #47

### CODE REVIEW

File: wfl/Models/EntityStatus.cs [hunk 1]
```diff
@@ -10,7 +10,8 @@ public class EntityStatus
         public enum Status
```
Comment (enricoferrari) [inline wfl/Models/EntityStatus.cs]: I don't foresee this status being a generic thing. So I would name it specific to orders. I would be curious to hear others opinions on this but what do you guys think of defining the enum inside the FoodOrder vs in it's own top level file?

File: wfl/Controllers/FoodOrdersController.cs [hunk 2]
```diff
@@ -68,5 +68,19 @@ public FoodOrdersController(IDBManager dbManager)
             order.setStatus(EntityStatus.Status.Active);
             return Ok(await dbManager.UpdateEntityAsync(id, order));
         }
+
+        [Route("{id}/complete")]
+        [HttpPost]
+        public async Task<IHttpActionResult> CompleteAsync([FromUri] string id)
+        {
+            FoodOrder order = await dbManager.GetEntityByIdAsync<FoodOrder>(id);
+            if (order.Status != EntityStatus.Status.Active)
```
Comment (enricoferrari) [inline wfl/Controllers/FoodOrdersController.cs]: you probably also want to null check order here

**Review by enricoferrari — CHANGES_REQUESTED** [review]

File: wfl/Controllers/FoodOrdersController.cs (hunk 2 above)
Comment (TNagaraja) [reply wfl/Controllers/FoodOrdersController.cs]: Changed in latest code

File: wfl/Models/EntityStatus.cs (hunk 1 above)
Comment (TNagaraja) [reply wfl/Models/EntityStatus.cs]: I'll move the status inside the FoodOrder class.

File: wfl/Controllers/FoodOrdersController.cs [hunk 3]
```diff
@@ -68,5 +68,19 @@ public FoodOrdersController(IDBManager dbManager)
             order.setStatus(EntityStatus.Status.Active);
             return Ok(await dbManager.UpdateEntityAsync(id, order));
         }
+
+        [Route("{id}/complete")]
+        [HttpPost]
+        public async Task<IHttpActionResult> CompleteAsync([FromUri] string id)
```
Comment (enricoferrari) [inline wfl/Controllers/FoodOrdersController.cs]: I think @jspringer2020 mentioned this in another PR, but we could follow suit here and make this a generic controller method to set the status. The interesting part comes in the profile. There you advertise certain actions based on what the current status is. So IF an orders status is `Active` the profile will show a `complete` action. That action will call this method with the status `Completed`.

Comment (TNagaraja) [reply wfl/Controllers/FoodOrdersController.cs]: I was in the process of making this change - should be up soon.

Comment (TNagaraja) [reply wfl/Controllers/FoodOrdersController.cs]: Changed now.

**Review by enricoferrari — APPROVED** [review]

**Review by jspringer2020 — APPROVED** [review]
Looks good. This is a good starting point as we talked IRL yesterday. Longer term we would likely use a validation framework to handle a lot of this for us.

**Pull request merged**